src/theory/quantifiers/rewrite_verifier.cpp

```diff
@@ -0,0 +1,137 @@
+/******************************************************************************
+ * Top contributors (to current version):
+ *   Andrew Reynolds
+ *
+ * This file is part of the cvc5 project.
+ *
+ * Copyright (c) 2009-2023 by the authors listed in the file AUTHORS
+ * in the top-level source directory and their institutional affiliations.
+ * All rights reserved.  See the file COPYING in the top-level source
+ * directory for licensing information.
+ * ****************************************************************************
+ *
+ * A class for finding unsoundness in the rewriter
+ */
+
+#include "theory/quantifiers/rewrite_verifier.h"
+
+#include "smt/env.h"
+#include "expr/node_algorithm.h"
+
+namespace cvc5::internal {
+namespace theory {
+namespace quantifiers {
+
+RewriteVerifier::RewriteVerifier(Env& env) : ExprMiner(env) {}
+
+bool RewriteVerifier::addTerm(Node n, std::vector<Node>& rewrites)
+{
+  Node nr = rewrite(n);
+  if (!checkEquivalent(n, nr))
+  {
+    std::stringstream ss;
+    checkEquivalent(n, nr, &ss);
+    Warning() << ss.str();
```

Comment:
It seems wasteful to call `checkEquivalent` a second time just to get the warning output. So, maybe get the warning from the first check? Also, since `Warning()` is used only once here, can we move it to `checkEquivalent`?

Reply:
It is highly expected that this code is never run, since we extremely rarely have bugs in our rewriter.  The reason we don't call it only once it because there is overhead in allocating the std::stringstream.